### ISSUE TITLE
GROUP BY returns zero over range, even though series contains non-zero values

### ISSUE THREAD

**Issue by victorhooi** [influxdb/influxdb#3240]
I have a series called "read_requests".

It contains values - e.g.:
```
> select value from "read_requests" WHERE time > '2015-07-01 08:21:00' AND time < '2015-07-01 08:21:10' AND device = 'sdb'
name: read_requests
-------------------
time                    value
2015-07-01T08:21:01Z    139.00
2015-07-01T08:21:02Z    70.00
2015-07-01T08:21:03Z    236.00
2015-07-01T08:21:04Z    177.00
2015-07-01T08:21:05Z    149.00
2015-07-01T08:21:06Z    126.00
2015-07-01T08:21:07Z    144.00
2015-07-01T08:21:08Z    148.00
2015-07-01T08:21:09Z    92.00
```
However, when I do a `mean(value)`, grouping by 1-second over that period, it gives me a mean of zero for *every* second:
```
> select mean(value) from "read_requests" WHERE time > '2015-07-01 08:21:00' AND time < '2015-07-01 08:21:10' AND device = 'sdb' GROUP BY time(1s)
name: read_requests
-------------------
time                    mean
2015-07-01T08:21:00Z
2015-07-01T08:21:01Z    0
2015-07-01T08:21:02Z    0
2015-07-01T08:21:03Z    0
2015-07-01T08:21:04Z    0
2015-07-01T08:21:05Z    0
2015-07-01T08:21:06Z    0
2015-07-01T08:21:07Z    0
2015-07-01T08:21:08Z    0
2015-07-01T08:21:09Z    0
```
Is that expected?

Also, when I do a `mean(value)` on the entire series, it also gives me zero - and the time is also weird, and doesn't seem to have any relationship to either the specified start or end time:
```
> select mean(value) from "read_requests" WHERE time > '2015-07-01 08:21:00' AND time < '2015-07-01 08:21:10' AND device = 'sdb'
name: read_requests
-------------------
time                            mean
2015-07-01T08:20:52.852172Z     0
```
Any ideas on what's going on?

**Comment by victorhooi** [comment]
Simplest reproduction:

* Python script to load in the data
```
#!/usr/bin/env python3
from influxdb import InfluxDBClient

client = InfluxDBClient(host='localhost', port=8086, database="influx_3240")

json_points = [{
        'measurement': 'read_requests',
        'tags': {'device': 'sda', 'hostname': 'iapp80'},
        'time': '2015-06-29T19:21:22-04:00',
        'fields': {'value': '3.98'},
}]

client.write_points(json_points)
```
* Shell commands to get `mean(value)`:
```
> create database influx_3240
> use influx_3240
Using database influx_3240
> show series
name: read_requests
-------------------
_key                                            device  hostname
read_requests,device=sda,hostname=iapp80        sda     iapp80

> select * from read_requests
name: read_requests
tags: device=sda, hostname=iapp80
time                    value
----                    -----
2015-06-29T23:21:22Z    3.98

> select mean(value) from read_requests
name: read_requests
-------------------
time                    mean
1970-01-01T00:00:00Z    0
```

**Comment by beckettsean** [comment]
@victorhooi The behavior suggests that your values are actually writing as strings. I don't grok python enough to know, but if your write statements are actually inserting `'value'='3.98'` then you are storing that value as the string "3.98", not a floating point value. Numbers, whether floats or integers, must be unquoted.

**Comment by victorhooi** [comment]
@beckettsean You're absolutely right =) - yeah, the issue was I had to explicitly cast it to a `float`.

I was a bit confused, because Grafana plotted the raw `value` field in a graph fine - it was only when I tried to graph `max(value)` that I got all zeroes:

https://github.com/grafana/grafana/issues/2278#issuecomment-118691564

Perhaps grafana has some functionality to explicitly cast `value` to a number, but not other things?

And from the `influx` shell, when I did `select`, there weren't any quotes or `str()` around the number, nor any error message when I tried to do numerical operations around it (just a result of zero).

I guess I'd stupidly assumed that InfluxDB was doing some magic to automatically parse the format of the field. Thanks for confirming =).
